### PR TITLE
Require Numba 0.57.0+ & NumPy 1.21.0+

### PR DESCRIPTION
**Pull request by jakirkham**
Align with the rest of RAPIDS on these requirements. Also needed for CUDA 12 support.

### CODE REVIEW

File: docs/source/install.rst [hunk 1]
```diff
@@ -64,7 +64,7 @@ Test Dependencies
 
     conda install -n ucx -c rapidsai -c nvidia -c conda-forge \
         pytest pytest-asyncio \
-        cupy "numba>=0.46" rmm \
+        cupy "numba>=0.57" rmm \
         distributed
```
Comment (bdice) [inline docs/source/install.rst:68]: This list of testing dependencies doesn't align with the contents here: https://github.com/rapidsai/ucx-py/blob/6f7c1aec5889eb08e488a3c9e85cdf5ad25e1368/dependencies.yaml#L73-L81

Differences are in rmm/cudf and version pinnings for numba.

File: pyproject.toml [hunk 2]
```diff
@@ -31,7 +31,7 @@ authors = [
 license = { text = "BSD-3-Clause" }
 requires-python = ">=3.9"
 dependencies = [
-    "numpy",
+    "numpy >=1.21",
     "pynvml >=11.4.1",
```
Comment (bdice) [inline pyproject.toml:35]: `pynvml` isn't listed as a [run dependency](https://github.com/rapidsai/ucx-py/blob/6f7c1aec5889eb08e488a3c9e85cdf5ad25e1368/dependencies.yaml#L67-L72) in `dependencies.yaml`, but `ucx` is listed? We need to make sure this is using `rapids-dependency-file-generator`.

**Review by bdice — CHANGES_REQUESTED** [review]
This needs to update `dependencies.yaml` and check on `rapids-dependency-file-generator`.

File: docs/source/install.rst (hunk 1 above)
Comment (jakirkham) [reply docs/source/install.rst:68]: Good point. This should now be fixed. Please let me know what you think :)

File: pyproject.toml (hunk 2 above)
Comment (jakirkham) [reply pyproject.toml:35]: Not sure what's going on there

The recipe [loads dependencies from `pyproject.toml`]( https://github.com/rapidsai/ucx-py/blob/6f7c1aec5889eb08e488a3c9e85cdf5ad25e1368/conda/recipes/ucx-py/meta.yaml#L36-L38 ) atm

Maybe there is a better way to make this more generally consistent

Comment (bdice) [reply pyproject.toml:35]: I'm fixing this and will file a PR shortly. It may supersede this PR if that's okay with you.

Comment (bdice) [reply pyproject.toml:35]: Opened this: https://github.com/rapidsai/ucx-py/pull/952

Still working on it but it's close to complete, I think.

**Comment by jakirkham** [comment]
Superseded by PR ( https://github.com/rapidsai/ucx-py/pull/952 )

Thanks Bradley! 🙏

**Pull request closed without merging**